tinker/faculty_bios/faculty_bio_controller.py

```diff
@@ -122,21 +124,201 @@ def _iterate_child_xml(self, child, author):
             school_array = []
             for school in child.findall('.//job-titles/school'):
                 school_array.append(school.text or 'Other')
-
-            page_values = {
-                'author': child.find('author') or None,
-                'id': child.attrib['id'] or "",
-                'title': child.find('title').text or None,
-                'created-on': child.find('created-on').text or None,
-                'path': 'https://www.bethel.edu' + child.find('path').text or "",
-                'schools': school_array,
-                'last-name': child.find('.//last').text or None,
-                'deactivated': child.find('.//deactivate').text or None
+            if not csv:
+                page_values = {
+                    'author': child.find('author') or None,
+                    'id': child.attrib['id'] or "",
+                    'title': child.find('title').text or None,
+                    'created-on': child.find('created-on').text or None,
+                    'path': 'https://www.bethel.edu' + child.find('path').text or "",
+                    'schools': school_array,
+                    'last-name': child.find('.//last').text or None,
+                    'deactivated': child.find('.//deactivate').text or None
             }
+            else:
+
+                # Creates a BeautifulSoup object
+                soup = BeautifulSoup(self.element_tree_to_html(child))
+                # Author isn't always filled out so we need to check if it is
+                if not soup.find('author'):
+                    author_text = " "
+                else:
+                    author_text = soup.find('author').text
+
+                # max_jobs will hold the maximum number of jobs someone has
+                max_jobs = 0
+                # my_list is a list used to add both job and education dictionaries to
+                my_list = []
+                # Dictionary holds each job-titles subfield
+
+                # Logic used to group together department/program and department chair/lead faculty/program director/
+                # job_title
+                for jobs in soup.findAll('job-titles'):
+                    max_jobs += 1
+                    school = jobs.find('school').text
+                    department = ""
+                    title = ""
+                    if school == 'Bethel University':
+                        department = ""
+                        title = jobs.find('job_title').text
```

Comment:
you have the line `title = jobs.find('job_title').text` duplicated everywhere throughout this section. I think you can rework this to make that work.

Reply:
Got rid of duplicated code.